Sources/ArgumentParser/Parsable Types/ParsableArgumentsValidation.swift

```diff
@@ -185,3 +186,59 @@ struct ParsableArgumentsCodingKeyValidator: ParsableArgumentsValidator {
     }
   }
 }
+
+/// Ensure argument names are unique within a `ParsableArguments` or `ParsableCommand`.
+struct ParsableArgumentsUniqueNamesValidator: ParsableArgumentsValidator {
+  struct Error: Swift.Error, CustomStringConvertible {
+    var duplicateNames: [String: Int] = [:]
+
+    var occurred: Bool {
+      !duplicateNames.isEmpty
+    }
+
+    var description: String {
+      duplicateNames.map { entry in
+        "Multiple (\(entry.value)) `Option` or `Flag` arguments are named \"\(entry.key)\"."
+      }.joined(separator: "\n")
+    }
+  }
+
+  static func validate(_ type: ParsableArguments.Type) throws {
+    let argSets: [ArgumentSet] = Mirror(reflecting: type.init())
+      .children
+      .compactMap { child in
+        guard
+          var codingKey = child.label,
+          let parsed = child.value as? ArgumentSetProvider
+          else { return nil }
+
+        // Property wrappers have underscore-prefixed names
+        codingKey = String(codingKey.first == "_" ? codingKey.dropFirst(1) : codingKey.dropFirst(0))
+
+        let key = InputKey(rawValue: codingKey)
+        return parsed.argumentSet(for: key)
+    }
+
+    let countedNames: [String: Int] = argSets.reduce(into: [:]) { countedNames, args in
+      switch args.content {
+      case .arguments(let defs):
+        for name in defs.flatMap({ $0.names }) {
+          countedNames[name.valueString, default: 0] += 1
+        }
+      default:
+        break
+      }
+    }
+
+    var error = Error()
+
+    let duplicateNames = countedNames.filter { $0.value > 1 }
+    if !duplicateNames.isEmpty {
+      error.duplicateNames = duplicateNames
+    }
+
+    if error.occurred {
+      throw error
+    }
```

Comment:
How about creating and throwing the error in `if !duplicateNames.isEmpty { ... }`? Then we can remove `error.occurred` which is actually `!duplicateNames.isEmpty`.

```
if !duplicateNames.isEmpty {
  throw Error(duplicateNames: duplicateNames)
}
```

Reply:
Yes, thank you. This is another remnant of `Error` having been a more complicated type. I really needed other sets of eyes on this.

Reply:
Done.